### ISSUE TITLE
Texture2D is not readable

### ISSUE THREAD

**Issue by BigGiantHead** [SrejonKhan/AsyncImageLibrary#4]
I went through the code and at line 76 you are making the texture not readable. Is there a reason for this?
It is making it difficult working with the texture.

**Comment by SrejonKhan** [comment]
Yes, it was used for free memory. However, I never thought of it that way. I guess it would be better to make the texture readable. Perhaps, an optional flag in the constructor to determine whether make the texture readable or not after processing. 

If you have made changes, please open a Pull Request. I would be happy to merge your changes. Thanks.

**Comment by BigGiantHead** [comment]
Sure, I get it. All right, I will make a pull request once I make my changes. I will try to add a parameter for this.

**Comment by SrejonKhan** [comment]
@BigGiantHead Gentle ping! Would you like to open a PR with your changes? Your contribution would be greatly appreciated.
